pr:
  - master
  - releases/*

jobs:
- template: run-tests.yml
  parameters:
    name: Py312_Ubuntu
    vmImage: 'ubuntu-latest'
    pythonVersion: '3.12'
# For some reason on Azure, 3.12 has an issue with pycryptodomex
#- template: run-tests.yml
#  parameters:
#    name: Py312_Mac
#    vmImage: 'macOS-latest'
#    pythonVersion: '3.12'
- template: run-tests.yml
  parameters:
    name: Py312_Win
    vmImage: 'windows-latest'
    pythonVersion: '3.12'
- template: run-tests.yml
  parameters:
    name: Py311_Ubuntu
    vmImage: 'ubuntu-latest'
    pythonVersion: '3.11'
- template: run-tests.yml
  parameters:
    name: Py311_Mac
    vmImage: 'macOS-latest'
    pythonVersion: '3.11'
- template: run-tests.yml
  parameters:
    name: Py311_Win
    vmImage: 'windows-latest'
    pythonVersion: '3.11'
- template: run-tests.yml
  parameters:
    name: Py310_Ubuntu
    vmImage: 'ubuntu-latest'
    pythonVersion: '3.10'
- template: run-tests.yml
  parameters:
    name: Py310_Mac
    vmImage: 'macOS-latest'
    pythonVersion: '3.10'
- template: run-tests.yml
  parameters:
    name: Py310_Win
    vmImage: 'windows-latest'
    pythonVersion: '3.10'
- template: run-tests.yml
  parameters:
    name: Py39_Ubuntu
    vmImage: 'ubuntu-latest'
    pythonVersion: '3.9'
- template: run-tests.yml
  parameters:
    name: Py39_Mac
    vmImage: 'macOS-latest'
    pythonVersion: '3.9'
- template: run-tests.yml
  parameters:
    name: Py39_Win
    vmImage: 'windows-latest'
    pythonVersion: '3.9'
- template: run-tests.yml
  parameters:
    name: Py38_Ubuntu
    vmImage: 'ubuntu-latest'
    pythonVersion: '3.8'
- template: run-tests.yml
  parameters:
    name: Py38_Mac
    vmImage: 'macOS-latest'
    pythonVersion: '3.8'
- template: run-tests.yml
  parameters:
    name: Py38_Win
    vmImage: 'windows-latest'
    pythonVersion: '3.8'
